src/Microsoft.AspNetCore.Dispatcher/DependencyInjection/DispatcherServiceCollectionExtensions.cs

```diff
@@ -17,6 +17,14 @@ public static IServiceCollection AddDispatcher(this IServiceCollection services)
             }
 
             services.AddSingleton<IStartupFilter, DispatcherEndpointStartupFilter>();
+
+            services.AddSingleton<DispatcherBaseServices, DefaultDispatcherBaseServices>();
+            services.AddSingleton<HandlerFactoryBaseServices, DefaultHandlerFactoryBaseServices>();
+
+            services.ConfigureOptions<DefaultDispatcherConfigureOptions>();
+
+            services.AddTransient<IHandlerFactoryOptionsValueProvider, SimpleEndpointHandlerFactory>();
```

Comment:
Example, here's an endpoint factory that's always available. But it also comes from DI.